core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -460,6 +460,15 @@ public static ByteString concat(ByteString s0, ByteString s1) {
     return s0.concat(s1);
   }
 
+  /** SQL function CONCAT(arg, ...)*/
```

Comment:
I noticed that the number of function parameters in different databases is different:
Mysql 
```
CONCAT(string1,string2, ... );
```
Oracle
```
CONCAT(char1, char2)
```
PostgreSql
```
CONCAT(str_1, str_2, ...)
```

Reply:
I noticed this problem too, and I found that creating two SqlFunction for CONCAT is redundant and unnecessary, what's more, it maybe cause exception when look up method.
I have updated this pull request, concat function is equivalent to || operator.